Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverrideHandler.cpp

```diff
@@ -8,13 +8,34 @@
 
 #include <AzToolsFramework/API/ToolsApplicationAPI.h>
 #include <AzToolsFramework/Prefab/Overrides/PrefabOverrideHandler.h>
+#include <AzToolsFramework/Prefab/PrefabDomUtils.h>
 #include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
 #include <AzToolsFramework/Prefab/Undo/PrefabUndoRevertOverrides.h>
 
  namespace AzToolsFramework
 {
     namespace Prefab
     {
+        namespace Internal
+        {
+            // Helper to differentiate between adding/removing
+            // an entity object or adding/removing an entity property
+            bool IsDirectEntityPatch(PrefabDomConstReference patch)
+            {
+                auto pathMember = patch->get().FindMember("path");
+                if (pathMember != patch->get().MemberEnd())
+                {
+                    AZ::Dom::Path path(pathMember->value.GetString());
+                    if (path.Size() > 1)
+                    {
+                        return (path[path.Size() - 2] == PrefabDomUtils::EntitiesName);
```

Comment:
Would be helpful to have some comment.
Example direct entity patch path: `/Entities/Entity_[123213123123]`

Reply:
Added comment.